packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -557,6 +688,30 @@ export class StripeHelper {
     return subscriptions;
   }
 
+  /**
+   * Find and return a subscription for a customer of the given plan id.
+   *
+   * @param customer
+   * @param planId
+   */
+  findCustomerSubscriptionByPlanId(
+    customer: Stripe.Customer,
+    planId: string
+  ): Stripe.Subscription | undefined {
+    if (!customer.subscriptions) {
+      throw error.internalValidationError(
+        'findCustomerSubscriptionByPlanId',
+        {
+          customerId: customer.id,
+        },
+        'Expected subscriptions to be loaded.'
+      );
+    }
+    return customer.subscriptions.data.find(
+      (sub) => sub.items.data.find((item) => item.plan.id === planId) != null
```

Comment:
nit: `sub.items.data.some` would read better here.

Reply:
This method was just moved around, though wouldn't `some` get a list while this wants a single item?

Reply:
`some` just help with removing the ` != null` comparison as it already returns a boolean.  Like I said, nit. :D